numpy/lib/arraysetops.py

```diff
@@ -315,18 +322,87 @@ def reshape_uniq(uniq):
         return uniq
 
     output = _unique1d(consolidated, return_index,
-                       return_inverse, return_counts, equal_nan=equal_nan)
+                       return_inverse, return_counts,
+                       equal_nan=equal_nan, kind=kind)
     output = (reshape_uniq(output[0]),) + output[1:]
     return _unpack_tuple(output)
 
 
 def _unique1d(ar, return_index=False, return_inverse=False,
-              return_counts=False, *, equal_nan=True):
+              return_counts=False, *, equal_nan=True, kind=None):
     """
     Find the unique elements of an array, ignoring shape.
     """
     ar = np.asanyarray(ar).flatten()
 
+    integer_array = np.issubdtype(ar.dtype, np.integer)
+
+    if kind not in {None, 'sort', 'table'}:
+        raise ValueError(
+            f"Invalid kind: '{kind}'. Please use None, 'sort' or 'table'.")
+    
+    use_table_method = (
+        integer_array
+        and kind in {'table', None}
+        and not return_index
+        and not return_inverse
+        and not isinstance(ar, np.ma.MaskedArray)
+        and not ar.size == 0
+    )
+    if use_table_method:
+        try:
+            int(np.min(ar))
+        except TypeError:
+            # Not an integer. e.g.,
+            # a datetime dtype is a subdtype,
+            # but can't be used as an integer.
+            use_table_method = False
```

Comment:
OK, this is what I mentioned as an option for the `isin` PR. Slight nitpick: `datetime64` isn't an integer subdtype, only `timedelta64` is.

Reply:
I will submit a similar fix for `isin`, thanks!